include/fw_info.h

```diff
@@ -27,7 +27,30 @@ extern "C" {
 
 #define MAGIC_LEN_WORDS (CONFIG_FW_INFO_MAGIC_LEN / sizeof(u32_t))
 
-struct fw_info_abi;
+/**
+ * This struct is meant to serve as a header before a list of function pointers
+ * (or something else) that constitute the actual ABI. How to use the ABI, such
+ * as the signatures of all the functions in the list must be unambiguous for an
+ * ID/version combination.
+ */
+struct __packed fw_info_abi {
+	/* Magic value to verify that the struct has the correct format. */
+	u32_t magic[MAGIC_LEN_WORDS];
+
+	/* The id of the ABI. */
+	u32_t abi_id;
```

Comment:
Could we move the length field here? IMO it is good to be able to check the length without having much knowledge about the layout of the struct. It would simplify manual debugging for instance.

Reply:
Will be done in https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1378